pkg/ccl/streamingccl/logical/logical_replication_writer_processor.go

```diff
@@ -686,24 +503,24 @@ func (lrw *logicalReplicationWriterProcessor) flushBuffer(
 	}
 
 	if err := g.Wait(); err != nil {
-		return b.checkpoint, err
+		return err
 	}
 
 	flushTime := timeutil.Since(preFlushTime).Nanoseconds()
-	keyCount, byteCount := int64(len(b.buffer.curKVBatch)), flushByteSize.Load()
+	keyCount, byteCount := int64(len(kvs)), flushByteSize.Load()
 	lrw.debug.RecordFlushComplete(flushTime, keyCount, byteCount)
 
 	lrw.metrics.Flushes.Inc(1)
 	lrw.metrics.FlushHistNanos.RecordValue(flushTime)
 	lrw.metrics.FlushRowCountHist.RecordValue(keyCount)
 	lrw.metrics.FlushBytesHist.RecordValue(byteCount)
 	lrw.metrics.IngestedLogicalBytes.Inc(byteCount)
-	lrw.metrics.CommitLatency.RecordValue(timeutil.Since(b.buffer.minTimestamp.GoTime()).Nanoseconds())
-	lrw.metrics.IngestedEvents.Inc(int64(len(b.buffer.curKVBatch)))
 
-	releaseBuffer(b.buffer)
+	firstKeyTS := kvs[0].Value.Timestamp.GoTime()
+	lrw.metrics.CommitLatency.RecordValue(timeutil.Since(firstKeyTS).Nanoseconds())
```

Comment:
If we care a lot about this difference, we could have the batch return the min timestamp via batchStats and then record it on a per-batch (or chunk) basis.

Reply:
we’re grabbing here before we sort, which I think is the same as it was before?

Reply:
Ohhhhhh, I completely forgot we are sorting by ts.

Reply:
We’re not sorting by ts per se. we’re getting my the first key from the unsorted buffer before we sort it, which is thus still in rangefeed emit order.